xclim/core/bootstrapping.py

```diff
@@ -102,82 +108,108 @@ def bootstrap_func(compute_indice_func: Callable, **kwargs) -> xarray.DataArray:
                 per_key = name
             else:
                 da_key = name
-
     # Extract the DataArray inputs from the arguments
     da: DataArray = kwargs.pop(da_key)
-    per: Optional[DataArray] = kwargs.pop(per_key, None)
-    if per is None:
+    per_da: Optional[DataArray] = kwargs.pop(per_key, None)
+    if per_da is None:
         # per may be empty on non doy percentiles
         raise KeyError(
             "`bootstrap` can only be used with percentiles computed using `percentile_doy`"
         )
-
-    # List of years in base period
-    clim = per.attrs["climatology_bounds"]
-    per_clim_years = xarray.cftime_range(*clim, freq="YS").year
-
-    # `da` over base period used to compute percentile
-    da_base = da.sel(time=slice(*clim))
-
-    # Arguments used to compute percentile
-    percentile = per.percentiles.data.tolist()  # Can be a list or scalar
+    # Boundary years of reference period
+    clim = per_da.attrs["climatology_bounds"]
+    if xclim.core.utils.uses_dask(da):
+        chunking = {d: "auto" for d in da.dims}
+        chunking["time"] = -1  # no chunking on time to use map_block
+        da = da.chunk(chunking)
```

Comment:
Is this auto-rechunking a performance improvement? I understand the `time = -1`, but passing `auto` for the other dimensions might override a user-intended chunking, adding unnecessary (or at least unwanted) steps in the dask tree?

Reply:
In any cases, I suggest raising a warning when the chunking is modified.

Reply:
The `"auto"` is useful to rechunk the whole thing and conserve chunk size to the configured `array.chunk-size` of dask.config.
I fear that without rechunking other dims, the chunks could become very large and dask workers might overflow their memory limits. This depends of course on the original chunking.

I agree a warning is necessary.
Plus, this step can be avoided altogether if time is already unchunked, I will add it as well.

Reply:
I'm not opposed to remove this "auto" rechunk but, it's a choice between:
- Preserving the overall chunk sizes to the configurable `array.chunk-size`.
- Preserving the chunked dimensions (other than time) to their original chunk sizes.

Reply:
Your suggestion of not rechunking if the time dimensions is already in a single block is, I think, the best choice! In the other case, I'm ok with an automatic rechunking  if a warning is issued.